contracts/TopShot.cdc

```diff
@@ -654,10 +667,184 @@ pub contract TopShot: NonFungibleToken {
+                "Jersey Number": "JerseyNumber",
+                "Draft Team": "DraftTeam",
+                "Draft Year": "DraftYear",
+                "Draft Selection": "DraftSelection",
+                "Draft Round": "DraftRound",
+                "Team At Moment": "TeamAtMoment",
+                "Primary Position": "PrimaryPosition",
+                "Height": "Height",
+                "Weight": "Weight",
+                "Total Years Experience": "TotalYearsExperience",
+                "NBA Season": "NbaSeason",
+                "Date Of Moment": "DateOfMoment",
+                "Play Category": "PlayCategory",
+                "Play Type": "PlayType",
+                "Home Team Name": "HomeTeamName",
+                "Away Team Name": "AwayTeamName",
+                "Home Team Score": "HomeTeamScore",
+                "Away Team Score": "AwayTeamScore"
+            }
+            for displayName in playMapping.keys {
+                let fieldName = playMapping[displayName]!
+                let fieldValue = TopShot.getPlayMetaDataByField(playID: self.data.playID, field: fieldName)
+                if fieldValue != nil {
+                    dict.insert(key: displayName, fieldValue)
+                }
+            }
+            return dict
+        }
+
+        // getMomentURL 
+        // Returns: The computed external url of the moment
+        pub fun getMomentURL(): String {
+            return "https://nbatopshot.com/moment/".concat(self.id.toString())
+        }
+        // getEditionName Moment's edition name is a combination of the Moment's setName and playID
+        // `setName: #playID`
+        pub fun getEditionName() : String {
+            let setName: String = TopShot.getSetName(setID: self.data.setID) ?? ""
+            let editionName = setName.concat(": #").concat(self.data.playID.toString())
+            return editionName
+        }
+
+        pub fun assetPath(): String {
+            return "https://assets.nbatopshot.com/media/".concat(self.id.toString())
+        }
+
+        // returns a url to display an medium sized image
+        pub fun mediumimage(): String {
+            let url = self.assetPath().concat("?width=512")
+            return self.appendOptionalParams(url: url, firstDelim: "&")
+        }
+
+        // a url to display a thumbnail associated with the moment
+        pub fun thumbnail(): String {
+            let url = self.assetPath().concat("?width=256")
+            return self.appendOptionalParams(url: url, firstDelim: "&")
+        }
+
+        // a url to display a video associated with the moment
+        pub fun video(): String {
+            let url = self.assetPath().concat("/video")
+            return self.appendOptionalParams(url: url, firstDelim: "?")
+        }
+
+        // appends and optional network param needed to resolve the media
+        pub fun appendOptionalParams(url: String, firstDelim: String): String {
+            let env = 0xNETWORK
```

Comment:
i think it makes sense to define these template parameters as top level variables so it doesn't get lost in the contract. At a glance everyone can know the template parameters needed to deploy the contract